common/src/Microsoft.Azure.IIoT.Agent.Framework/src/Agent/Default/WorkerSupervisor.cs

```diff
@@ -107,26 +117,34 @@ public class WorkerSupervisor : IWorkerSupervisor, IDisposable {
         /// </summary>
         /// <returns></returns>
         private async Task EnsureWorkersAsync() {
-            var workerStartTasks = new List<Task>();
 
-            while (true) {
-                var workers = _agentConfigProvider.Config?.MaxWorkers ?? kDefaultWorkers;
-                while (_instances.Count < workers) {
-                    _logger.Information("Creating new worker...");
-                    var worker = await CreateWorker();
-                }
-
-                foreach (var stoppedWorker in _instances.Keys.Where(s => s.Status == WorkerStatus.Stopped)) {
-                    _logger.Information("Starting worker '{workerId}'...", stoppedWorker.WorkerId);
-                    workerStartTasks.Add(stoppedWorker.StartAsync());
-                }
-                await Task.WhenAll(workerStartTasks);
-                // the configuration might have been changed by workers execution
-                var newWorkers = _agentConfigProvider.Config?.MaxWorkers ?? kDefaultWorkers;
-                if (workers >= newWorkers) {
-                    break;
-                }
+            if (_agentConfigProvider.Config?.MaxWorkers <= 0) {
```

Comment:
MaxWorkers can be null

Reply:
Nullable int can be safely checked against numbers, check will return false if property is null.

```c#
void Main()
{
	var cfg = new Config();
	cfg.MaxWorker = null;
	if (cfg?.MaxWorker <= 0) {
		Console.WriteLine("Should not hit");
	}
	else if (cfg?.MaxWorker > 0)
	{
		Console.WriteLine("Should not hit");
	}
	else {
		Console.WriteLine("As expected");
	}
}

class Config 
{
	public int? MaxWorker {get;set;}
}
```

Output:
```pwd
As expected
```

Reply:
Yes, in this case its clear what happens, but otherwise the implicit default behavior may lead to unexpected results, e.g. whatever happens when MaxWorkers > 0

Reply:
When MaxWorkers is bigger than 0, it was explicitly set by the customer (via API) and we have to create or terminate worker to reach the desired number